app/controllers/samples_controller.rb

```diff
@@ -138,14 +139,19 @@ def index_v2
                  current_power.samples
                end
 
-    respond_to do |format|
-      format.json do
-        render json: @samples.offset(offset).limit(limit).as_json(
-          only: [:id, :name, :sample_tissue, :host_genome_id, :project_id, :created_at],
-          methods: []
```

Comment:
this was calling the as_json method of Sample, which had been overridden with a method that makes a sql query for each record in @samples (see below)

Reply:
I am not sure what you mean... but this does not work exactly like that.
That `as_json` method only adds methods if you do not pass the parameter `methods` in. In this case is clearing it.

We might not need those methods by default, but that is another question... I am afraid that removing that method could affect something else on the website...

Reply:
I see. So I can remove `:input_files, :host_genome_name, :private_until` from my code here actually, since it was never present, due to `methods: []`.

Reply:
I'm inclined to take the risk of removing the `as_json` method from app/models/sample.rb because it could cause so many hidden problems. I will do a thorough audit to catch all the occurrences I can. If any issues come up with the removal after deployoment, I'll just fix them immediately

Reply:
Agree, i change the `as_json` recently(?) to allow at least overriding the methods arg and maintaining backward compatibility... but would be ideal to delete it and change any calls that require it.